miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -0,0 +1,319 @@
+        Group g = securityManager.getGroupById(group.getGroupId());
+        if (g == null) throw new IllegalArgumentException("Group " + group.getGroupId() + " does not exist");
+        resolved.add(g);
+      }
+    }
+    return resolved;
+  }
+
+  private long resolveMembers(SecurityProfile sp) throws IOException {
+    if (sp == null) throw new NullPointerException("null SecurityProfile");
+    sp.setOwner(securityManager.getUserById(sp.getOwner().getUserId()));
+    sp.setReadUsers(resolveUsers(sp.getReadUsers()));
+    sp.setReadGroups(resolveGroups(sp.getReadGroups()));
+    sp.setWriteUsers(resolveUsers(sp.getWriteUsers()));
+    sp.setWriteGroups(resolveGroups(sp.getWriteGroups()));
+    return securityProfileStore.save(sp);
+  }
+
+  private Collection<User> resolveUsers(Collection<User> users) throws IOException {
+    List<User> resolved = Lists.newArrayList();
+    if (users != null) {
+      for (User user : users) {
+        User u = securityManager.getUserById(user.getUserId());
+        if (u == null) throw new IllegalArgumentException("User " + user.getUserId() + " does not exist");
+        resolved.add(u);
+      }
+    }
+    return resolved;
+  }
+
+  @Override
+  public long saveProject(Project project) throws IOException {
+    authorizationManager.throwIfNotWritable(project);
+    ValidationResult shortNameValidation = namingScheme.validateProjectShortName(project.getShortName());
+    if (!shortNameValidation.isValid()) {
+      throw new IOException("Cannot save project - invalid shortName: " + shortNameValidation.getMessage());
+    }
+    if (project.getId() == ProjectImpl.UNSAVED_ID) {
+      resolveMembers(project.getSecurityProfile());
+      project.setName(generateTemporaryName());
+      projectStore.save(project);
+      try {
+        project.setName(namingScheme.generateNameFor(project));
+      } catch (MisoNamingException e) {
+        throw new IOException("Cannot save Project - issue with naming scheme", e);
+      }
+      LimsUtils.validateNameOrThrow(project, namingScheme);
+    } else {
+      Project original = projectStore.get(project.getId());
+      original.setAlias(project.getAlias());
+      original.setDescription(project.getDescription());
+      original.setIssueKeys(project.getIssueKeys());
+      original.setLastUpdated(new Date());
+      original.setProgress(project.getProgress());
+      original.setReferenceGenome(referenceGenomeDao.getReferenceGenome(project.getReferenceGenome().getId()));
+      original.setShortName(project.getShortName());
+      for (ProjectOverview po : project.getOverviews()) {
+        if (po.getId() == ProjectOverview.UNSAVED_ID) {
+          original.getOverviews().add(po);
+        }
+      }
+      updateSecurityProfile(original.getSecurityProfile(), project.getSecurityProfile());
+      project = original;
+    }
+    project.getSecurityProfile().setProfileId(saveSecurityProfile(project.getSecurityProfile()));
+    long id = projectStore.save(project);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "id". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)